packages/flame_3d/lib/src/components/component_3d.dart

```diff
@@ -1,48 +1,44 @@
-import 'dart:ui';
-
 import 'package:flame/components.dart' show Component, HasWorldReference;
-import 'package:flame/game.dart' show FlameGame;
 import 'package:flame_3d/camera.dart';
 import 'package:flame_3d/components.dart';
 import 'package:flame_3d/game.dart';
-import 'package:flame_3d/graphics.dart';
-import 'package:flame_3d/resources.dart';
 
 /// {@template component_3d}
-/// [Component3D]s are the basic building blocks for a 3D [FlameGame].
+/// [Component3D] is a base class for any concept that lives in 3D space.
 ///
 /// It is a [Component] implementation that represents a 3D object that can be
 /// freely moved around in 3D space, rotated, and scaled.
 ///
-/// The [Component3D] class has no visual representation of its own (except in
-/// debug mode). It is common, therefore, to derive from this class
-/// and implement a specific rendering logic.
+/// The main property of this class is the [transform] (which combines
+/// the [position], [rotation], and [scale]). Thus, the [Component3D] can be
+/// seen as an object in 3D space.
+///
+/// It is typically not used directly, but rather use one of the following
+/// implementations:
+/// - [Object3D] for a 3D object that can be bound and rendered by the GPU
+/// - [LightComponent] for a light source that affects how objects are rendered
+/// - [Group3D] for a generic container for several [Component3D]s
 ///
 /// The base [Component3D] class can also be used as a container
 /// for several other components. In this case, changing the position,
 /// rotating or scaling the [Component3D] will affect the whole
 /// group as if it was a single entity.
 ///
-/// The main property of this class is the [transform] (which combines
-/// the [position], [rotation], and [scale]). Thus, the [Component3D] can be
-/// seen as an object in 3D space where you can change its perceived
-/// visualization.
-///
-/// See the [MeshComponent] for a [Component3D] that has a visual representation
-/// by using [Mesh]es
 /// {@endtemplate}
-class Component3D extends Component with HasWorldReference<World3D> {
+abstract class Component3D extends Component with HasWorldReference<World3D> {
+  final Transform3D transform;
+
   /// {@macro component_3d}
   Component3D({
     Vector3? position,
     Vector3? scale,
     Quaternion? rotation,
-  }) : transform = Transform3D()
+    List<Component3D> children = const [],
+  })  : transform = Transform3D()
```

Comment:
Does this mean children can be positioned in 3D space relative to the parent? I don't think that will work right now without changes to t he `Transform3D` right?

Reply:
it should, but I don't think it would right now. this is exactly how it was before

Reply:
I agree it should be :P so one day we will be glad you already did this part of the code haha.